signing_clients/apps.py

```diff
@@ -102,6 +106,21 @@ def _digest(data):
     return {'md5': md5.digest(), 'sha1': sha1.digest()}
 
 
+def get_addon_type_from_signature(pkcs7):
+    info = SignatureInfo(pkcs7)
+    ou = get_signer_organizational_unit_name(info)
+
+    if ou == constants.OU_PRODUCTION:
+        # TODO: where should the CN be taken from to identify
+        # hotfix-addons? `if endEntity.CN = "firefox-hotfix@mozilla.org"`
```

Comment:
The `endEntity` is the certificate that matches the `issuerAndSerialNumber` value from the `SignerInfo` sequence of the signature. See https://github.com/mozilla/addons-server/pull/5868#discussion_r126283815 for the correct way to search for it.

In the listing below, `issuerAndSerialNumber` are listed at the bottom of the signature, and point to the certificate at the top.
![untitled](https://user-images.githubusercontent.com/415617/28116744-84ca1fa0-66d8-11e7-9914-cc8bfa9a1fab.png)